modules/SantoriniBoard.class.php

```diff
@@ -123,32 +132,14 @@ public function getPiecesByType($type, $type_arg = null, $location = null)
    * Order by tokens first, then by z-order ascending
    * parameter: send token visibles by this player
    */
-  public function getPlacedPieces($playerView= null)
-  {
-  
-    // add the abyss for Tartarus -- not the cleanest way
-    $abyss = false;
-    $tokenabyss = null;
-    $spaceabyss = null;
-    if ($playerView != null)
-    {
-      $powers = $this->game->playerManager->getPlayer($playerView)->getPowers();
-      if (count($powers)==1 && $powers[0]->getId() == TARTARUS && $powers[0]->getAbyssSpace() != null)
-      {  
-        $abyss = true;
-        $tokenabyss = $powers[0]->getToken();
-        $spaceabyss = $powers[0]->getAbyssSpace();
-        $spaceabyss['z'] = count($this->getBlocksAt($spaceabyss));
-        $this->setPieceAt($tokenabyss, $spaceabyss);
-      }
+  public function getPlacedPieces($playerId = null)
+  {
+    $sql = "SELECT * FROM piece WHERE location = 'board'";
+    if ($playerId != null) {
+      $sql .= " OR (location = 'secret' AND player_id = $playerId)";
```

Comment:
The hidden tokens now get placed in a separate board called "secret", maintaining the proper x, y, z coords. We only include these pieces from 'secret' board if you are the owner.